### ISSUE TITLE
Allow `conditional.defaultCase` to infer its return type

### ISSUE THREAD

**Issue by Brodzko** [remeda/remeda#605]
Is there a particular reason why `conditional.defaultCase()` automatically assumes its return type is `unknown`? I think it would be enough to just change its typing to 
```ts
const defaultCase: <In, Out>(then?: (data: In) => Out) => readonly [() => boolean, (data: In) => Out];
```
to allow for the same kind of inference we get from explicit cases. 

I find it kind of impractical that using `defaultCase` forces in an `unknown`, even though I'm intentionally using it as a "catchall" branch and _want_ to keep the type as part of the union returned from `conditional`.

**Comment by eranhirsch** [comment]
Thanks for reporting. I sent out a fix for this: #606

**Comment by Brodzko** [comment]
Damn that was a fast resolution, huge kudos and many thanks! 

Out of curiosity, is there a comprehensive contributing guide? I didn't find a `CONTRIBUTING.md`, browsed through some issues and it's apparent that great care is being taken when deciding which direction to take the API of `remeda`. 

I'm open to contribute if I happen to build an utility that I find useful in my day-job and think it might be useful "generally", but don't want to clutter/bother the maintainers.

Thanks!
